// nodework/lib/config.js
/**
 * Application global configs
 * 
 * @module config
**/

'use strict';

var dynamicFlag;

// set a log level - 'info', 'error', 'warn', 'debug'
exports.logLevel = 'error';

// list of valid file extensions to set mime type
exports.extensions = {
    '.html' : 'text/html',
    '.css' : 'text/css',
    '.js' : 'application/javascript',
    '.png' : 'image/png',
    '.gif' : 'image/gif',
    '.jpg' : 'image/jpeg',
    '.jpeg' : 'image/jpeg'
};

// object containing cross site information
exports.details = {
    'title' : 'NodeWork test site',
    'heading' : 'NodeWork Test',
    'subheading' : 'Node.js webserver using MVC style implementation'
};

exports.controllers = {
    'path' : process.cwd() + '/controllers/',
    'notfound' : '404notfound'
};

exports.models = {
    'path' : process.cwd() + '/models/'
};

exports.views = {
    'path' : process.cwd() + '/views/'
};

/**
 * This is where the layout templates are defined.  For now they're pretty simple layouts with
 * one dynamic section and multiple static sections.  The 'dynamic' value is defined by the controller
 * and must be present for now.  All other layout values must be present in the views folder
 */

// sets the global default layout identifier
exports.defaultLayout = 'default';

// sets the flag used to identify the dynamic part of the layout template
exports.dynamicFlag = dynamicFlag = 'dynamic';

// defines the different layout templates available
exports.layoutTemplates = {
    'default' : ['header', dynamicFlag, 'footer'],
    'sidebar' : ['header', 'sidebar', dynamicFlag, 'footer']
};
